### ISSUE TITLE
Firefox support

### ISSUE THREAD

**Issue by MPThLee** [shinycolors/hiori#10]
Since firefox doesn't support redirecting to `data:` scheme, It needs request modification.
In Firefox, [webRequest.filterResponseData()](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/webRequest/filterResponseData) supports request modification.

---
- [x] Support Firefox in code
- [ ] `applications` key in `manifest.json` ([MDN Docs](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/applications))
- [ ] Signing extension for normal use ([MDN Docs](https://developer.mozilla.org/en-US/Add-ons/Distribution))

**Comment by MPThLee** [comment]
Update: 'Support Firefox in code' resolved after 39472ac5fa

**Comment by dragonjet** [comment]
does signing mean publishing it to firefox add-ons store?

**Comment by MPThLee** [comment]
No, see also https://wiki.mozilla.org/Add-ons/Extension_Signing
[In MDN Docs](https://developer.mozilla.org/en-US/Add-ons/Distribution), it says
> Starting with Firefox 43, add-on extensions and multi-item add-ons that include extensions need to be signed by Mozilla before they can install in release and beta versions of Firefox. 

It means when extension signed, it can be install without developer mode.

ps. Normally extensions are automatically signed when upload to AMO(addons.mozilla.org)
